### PR TITLE
[FLINK-30279] Move existing RabbitMQ connector code from Flink repo to dedicated RabbitMQ repo

### PR DESCRIPTION
**Pull request by MartijnVisser**

### CODE REVIEW

**Review by zentol — CHANGES_REQUESTED** [review]
- missing production tests

**Comment by zentol** [comment]
Looks like this is again using the 1.17-SNAPSHOT code instead of 1.16.0.

**Comment by MartijnVisser** [comment]
> missing production tests

Fixed

> Looks like this is again using the 1.17-SNAPSHOT code instead of 1.16.0.

Also fixed

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -49,27 +63,39 @@ jobs:
...
+      - name: Compile and test flink-connector-rabbitmq
+        timeout-minutes: 30
+        run: |          
           set -o pipefail
 
           mvn clean deploy ${MVN_COMMON_OPTIONS} \
             -DaltDeploymentRepository=validation_repository::default::file:${{ env.MVN_VALIDATION_DIR }} \
             -Dscala-2.12 \
-            -Prun-end-to-end-tests -DdistDir=$(pwd)/../flink-1.16-SNAPSHOT \
+            -Prun-end-to-end-tests -DdistDir=${{ env.FLINK_CACHE_DIR }}/flink-${{ inputs.flink_version }} \
```
Comment (zentol) [inline .github/workflows/ci.yml]: will you merge this commit into the previous one? Most of these changes shouldn't be necessary.

**Review by zentol — CHANGES_REQUESTED** [review]
Can you rebase the PR against the current main?
I think we never set up the architecture tests properly in the CI scripts, and afaict this PR here should fail as-is.

**Comment by MartijnVisser** [comment]
> Can you rebase the PR against the current main?

Done

**Comment by MartijnVisser** [comment]
> I think we never set up the architecture tests properly in the CI scripts, and afaict this PR here should fail as-is.

Correct. I'm assuming that we want to want to ignore these errors for this PR?

**Comment by zentol** [comment]
> Correct. I'm assuming that we want to want to ignore these errors for this PR?

No. But you just need to run the architecture test locally and set...`-Dfreeze.refreeze=true`

**Review by zentol — APPROVED** [review]

**Comment by zentol** [comment]
Please remember to address https://github.com/apache/flink-connector-rabbitmq/pull/3#discussion_r1045606463

**Pull request merged**